oqs/oqs.py

```diff
@@ -37,13 +45,48 @@ def _load_shared_obj(name):
     raise RuntimeError("No " + name + " shared libraries found")
 
 
+def _install_liboqs(directory):
+    """Attempts to install liboqs automatically."""
+    oqs_install_str = (
+        "cd "
+        + directory
+        + """
+git clone https://github.com/open-quantum-safe/liboqs --depth 1
+cmake -S liboqs -B liboqs/build -DBUILD_SHARED_LIBS=ON
+cmake --build liboqs/build --parallel 4
+"""
+    )
+    cmake_install_cmd = "cmake --build liboqs/build --target install"
+    if platform.system() == "Windows":
+        raise RuntimeError(
+            "Automatic installation of liboqs is not (yet) supported on Windows"
+        )
+    else:
+        oqs_install_str += "sudo " + cmake_install_cmd
```

Comment:
Is forcing `sudo` as part of an install good practice? What about leaving this away thus making the user aware of the need for superuser rights? Or alternatively, wouldn't it be possible to install `liboqs` to a non-root-owned installation location (the one where `liboqs-python` goes to)? But I'm not enough of a "Python guy" to best comment on this installation approach. Also, what if [venv](https://python.land/virtual-environments/virtualenv) is used?